supported_platforms.md

```diff
@@ -0,0 +1,51 @@
+# Microsoft Azure IoT SDKs for .NET
+
+This SDK is tested nightly on a mix of .NET frameworks on both Windows 10 and on Ubuntu 1604. For additional details
+for each tested platform, see the respective sections below. 
+
+## Supported .NET versions
+
+The NuGet packages provide support for the following .NET versions:
+- .NET Standard 2.1
+- .NET Standard 2.0
+- .NET Framework 4.7.2 (IoT Hub SDKs only)
+- .NET Framework 4.5.1 (IoT Hub SDKs only)
+
+This SDK _may_ work with other versions of .NET, but there are no guarantees that they will _always_ work for those 
```

Comment:
This isn't the right way to think about this.

.NET Standard 2.0 and 2.1 are "standards" not frameworks. A user wouldn't target them. Rather they target a version of .NET that is compatible with these standards and, based on framework loading preferences, gets one of these.

Secondly, if a user us running 4.8, the [framework loading precedence](https://nugettools.azurewebsites.net/5.5.0-preview.2.6382/framework-precedence?framework=net4.8&exludedIdentifiers=) will choose 4.7.2. That doesn't mean we won't support them. My take is that is fully supported by .NET and us.

What are your thoughts?

Reply:
> Secondly, if a user us running 4.8, the framework loading precedence will choose 4.7.2. That doesn't mean we won't support them. My take is that is fully supported by .NET and us.

This SDK isn't really responsible for defining how framework loading precedence works, so I'm not convinced we need to outline that here.

Reply:
> .NET Standard 2.0 and 2.1 are "standards" not frameworks. A user wouldn't target them. Rather they target a version of .NET that is compatible with these standards and, based on framework loading preferences, gets one of these.

Would you rather I list the .NET Core and .NET framework versions we support since those are the .NET versions users will actually target when building their applications?

Reply:
No. I think listing our targets are fine. I just think we shouldn't say 'there are no guarantees that they will always work".

The SDK isn't responsible for framework loading precedence, but the .NET framework for the app they are running is. MS supports having a newer app load an older library. We shouldn't make it sound like they have to run an app with the target we have, or it might not work or we might not support them.

That is my point.

Reply:
The iffy versions here are the .NET framework versions that we support. Technically, since our lib is ns2.0 compatible, all implementations of ns2.0+ should work fine. If it doesn't, that's usually a bug in our code => the readonly dict implementation that we had handwritten, and was broken for .NET 5.0.

How about we call out the .NET framework versions saying that they'll pick the net451/ net472 dll as per framework precedence. Anything ns2.0+ should always work.

Reply:
sure, try that.

Reply:
Now that we have the below explanation, we can remove this statement. Based on the .NET dll targeted, their app should always work if the API support is available

Reply:
```suggestion
This SDK _may_ work with newer versions of .NET, but there are no guarantees that they will _always_ work for those until we officially add support for them.
```